platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/DelayedExecutor.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Waits for a set of predicates before executing a given function.
+ * <p>
+ * This class periodically checks all given predicates and executes the given
+ * function only if all predicates are true.
+ */
+class DelayedExecutor {
```

Comment:
Interesting idea. I looked into it, but since I am no expert with JavaScript-Promises, and the little time we have, I would leave that for a future improvement.

Reply:
ok